org-code-javabuilder/lib/src/main/java/org/code/javabuilder/InternalFacingException.java

```diff
@@ -0,0 +1,22 @@
+package org.code.javabuilder;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+/**
+ * Exception intended to eventually bubble up to a logger (i.e. CloudWatch) but not to the user.
+ * These should generally be used after the user's code has finished running.
+ */
+public class InternalFacingException extends Exception {
+  public InternalFacingException(String errorMessage) {
+    super(errorMessage);
+  }
+
+  /** @return A pretty version of the exception and stack trace. */
+  public String getLoggingString() {
+    StringWriter stringWriter = new StringWriter();
```

Comment:
Do we expect to add more context to the logging string in the future?  (I forget, but I think one of the built-in Exception methods includes context and stack trace.)

Reply:
Context as in CPU usage etc.?

Reply:
Sorry, ignore this comment.  I was referring to just the message and name of the exception but it looks like  `printStackTrace` includes those already.